core-backup/src/test/scala/io/aiven/guardian/kafka/backup/MockedBackupClientInterface.scala

```diff
@@ -80,19 +80,18 @@ class MockedBackupClientInterface(override val kafkaClientInterface: MockedKafka
     */
   override type BackupResult = Done
 
-  override type CurrentState = Nothing
+  override type State = Nothing
 
-  override def getCurrentUploadState(key: String): Future[Option[Nothing]] = Future.successful(None)
+  override def getCurrentUploadState(key: String): Future[UploadStateResult] =
+    Future.successful(UploadStateResult.empty)
 
   override def empty: () => Future[Done] = () => Future.successful(Done)
 
-  /** Override this method to define how to backup a `ByteString` to a `DataSource`
```

Comment:
This doc isn't doing anything useful since its already documented in `BackupClientInterface` and its also a test we are dealing with.